### PR TITLE
add nix flake

### PR DESCRIPTION
**Pull request by undefined-moe**
# Pull Request Template

## Description

This pull request adds flake.nix, to simplify packaging process.
Use `nix bundle --bundler github:ralismark/nix-appimage .#ecli-rs`, `nix bundle --bundler github:ralismark/nix-appimage .#ecli-server` to create a static linked AppImage file.

NOTE: this requires `ecli/Cargo.lock` exists and added into git tree (`git add ecli/Cargo.lock -f`), but no need to commit it.

P.S: According to [Cargo Book](https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html#:~:text=If%20you%E2%80%99re%20building%20an%20end%20product%2C%20which%20are%20executable%20like%20command%2Dline%20tool%20or%20an%20application%2C%20or%20a%20system%20library%20with%20crate%2Dtype%20of%20staticlib%20or%20cdylib%2C%20check%20Cargo.lock%20into%20git), "If you’re building an end product, which are executable like command-line tool or an application, or a system library with crate-type of staticlib or cdylib, check Cargo.lock into git."

I've tested that `nix bundle --bundler github:NixOS/bundlers#toDockerImage .#ecli-rs` also works well, producing a ~27M docker image.

See also: #258

## Type of change

New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

N/A

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by oluceps** [comment]
Hi, Thanks for your contributions!
You could consider continuing on this branch https://github.com/eunomia-bpf/eunomia-bpf/tree/flake-devenv for avoiding duplicate works.

And some further plans and infomations about nix integration on `eunomia-bpf`: #158

**Comment by yunwei37** [comment]
any progress on this pr？

File: flake.nix
Comment (ocfox) [inline flake.nix:42]: openssl, elfutils and zlib should belong to buildInputs.

File: flake.nix
Comment (ocfox) [inline flake.nix:52]: Don't forget the runHook.

File: flake.nix
Comment (ocfox) [inline flake.nix:75]: runHook

File: flake.nix
Comment (ocfox) [inline flake.nix:113]: runHook

**Review by ocfox — APPROVED** [review]

**Comment by ocfox** [comment]
I clicked approve by mistake, please ignore it.

**Review by ocfox — APPROVED** [review]
LGTM now, wait for ci tests.

**Comment by ocfox** [comment]
Before merging, some conflicts need to be resolved.

**Pull request merged**
